kong/tools/queue.lua

```diff
@@ -506,6 +514,21 @@ local function enqueue(self, entry)
     return nil, "entry must be a non-nil Lua value"
   end
 
+  if self.concurrency_limit == -1 then -- unlimited concurrency
+    -- do not enqueue when concurrency_limit is unlimited
+    local ok, err = timer_at(0, function(premature)
```

Comment:
Found some places use `kong.timer`, so we do not need to declare `ngx.timer.at` spearately.